app/models/labware_creators/common_file_handling/csv_file_base.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require './lib/nested_validation'
+require 'csv'
+
+# Part of the Labware creator classes
+module LabwareCreators
+  #
+  # This is an abstract class for handling csv files.
+  #
+  class CommonFileHandling::CsvFileBase
+    include ActiveModel::Validations
+    extend NestedValidation
+
+    validate :correctly_parsed?
+
+    def initialize(file)
+      initialize_variables(file)
+    rescue StandardError => e
+      reset_variables
+      @parse_error = e.message
+    ensure
+      file.rewind
+    end
+
+    # Override in subclass if needed
+    def initialize_variables(file)
+      @filename = file.original_filename
+      @data = CSV.parse(file.read)
+      remove_bom
+      @parsed = true
+    end
+
+    # Override in subclass if needed
+    def reset_variables
+      @parent_barcode = nil
```

Comment:
Not sure why `@parent_barcode` is being reset, do you know? The other 3 are initialized in this class.

Reply:
well spotted, removed
cut and paste error from the pcr cycles version of csv file base